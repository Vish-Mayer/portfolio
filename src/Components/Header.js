import React from 'react';

function Header() {
  return(
  <div>
    <h1>Vishal Mayer Kakkad - Full Stack Developer</h1>
  </div>
  )
}

export default Header
