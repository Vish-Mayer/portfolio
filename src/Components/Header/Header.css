
.big-banner{
  background-image: url('../../Assets/Other/background1.jpg');
  height: 100vh;
  background-size: cover;
  background-position: center;
  border-radius: 0;
  display: table;
}

.intro-message{
  position: absolute;
  color: whitesmoke;
  max-height: 100%;
  max-width: 100%;
  padding-top: 20em;
  padding-left: 5em;
}

.fa-chevron-down{
  position: absolute;
  bottom: 0em;
}

@media only screen and (max-width: 650px) {
  .intro-message {
    padding-top: 14em;
    padding-left: 1em;
  }
}

@media only screen and (max-height: 490px) {
  .intro-message {
    padding-top: 0em;
    padding-left: 1em;
  }
}


