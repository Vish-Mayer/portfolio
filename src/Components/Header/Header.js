import React from 'react';
import Typical from 'react-typical';
import { HashLink as Link } from 'react-router-hash-link';

import './Header.css'
import icon from '../../Assets/dwn-arrowXS.png'

function Header() {

  return(
  <div className="header big-banner container-fluid d-flex " id="home">
    <div className="intro-message d-flex overflow">
      <div className="row">
        <div class="w-100">
          <h2>
            Hi I'm Vish, I'm a Full Stack Developer.
            <p/>I write applications in {' '}
            <Typical
                steps={['JavaScript.', 2000, 'React.', 2000, 'Ruby.', 2000, 'Ruby on Rails.', 2000 ]}
                loop={Infinity}
                wrapper="b"
              />
            </h2>
        </div>
      </div>  
    </div>
      <div class="container-fluid d-flex justify-content-center">                  
        <Link className="dwn-btn" smooth to="/#about">
          <img src={icon}/>
        </Link>
    </div>
  </div>
  
  )
}

export default Header
