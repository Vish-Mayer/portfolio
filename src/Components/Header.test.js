import React from 'react';
import Typical from 'react-typical'
import Header from './Header';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Header', () => {

  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Header/>)
  });

  it('renders title of application', () => {
    const intro = ("Hi Im Vish, I'm a Full Stack Developer,I write applications in  ")
    expect(wrapper.find("h2").text()).toContain(intro)
  });

  it('loops through a list of languages', () => {
    const output = wrapper.find("h2").props().children
    expect(JSON.stringify(output)).toMatchSnapshot();
  });
});
