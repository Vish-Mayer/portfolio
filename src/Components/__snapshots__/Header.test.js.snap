// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header loops through a list of languages 1`] = `"[\\"Hi Im Vish, I'm a Full Stack Developer,\\",{\\"type\\":\\"p\\",\\"key\\":null,\\"ref\\":null,\\"props\\":{},\\"_owner\\":null,\\"_store\\":{}},\\"I write applications in \\",\\" \\",{\\"type\\":{\\"compare\\":null},\\"key\\":null,\\"ref\\":null,\\"props\\":{\\"steps\\":[\\"JavaScript\\",2000,\\"React\\",2000,\\"Ruby\\",2000,\\"Ruby on Rails\\",2000],\\"loop\\":null,\\"wrapper\\":\\"b\\"},\\"_owner\\":null,\\"_store\\":{}}]"`;
